notebooks/search/06-synonymys-api.ipynb

```diff
@@ -0,0 +1,517 @@
+        "id": "4AXB9IR8JjCT",
+        "outputId": "31d59878-88a8-4294-a727-0271d3890e1c"
+      },
+      "outputs": [],
+      "source": [
+        "settings = {\n",
+        "    \"analysis\": {\n",
+        "        \"analyzer\": {\n",
+        "            \"my_custom_index_analyzer\": {\n",
+        "                \"tokenizer\": \"standard\",\n",
+        "                \"filter\": [\n",
+        "                    \"lowercase\"\n",
+        "                ]\n",
+        "            },\n",
+        "            \"my_custom_search_analyzer\": {\n",
+        "                \"tokenizer\": \"standard\",\n",
+        "                \"filter\": [\n",
+        "                    \"lowercase\",\n",
+        "                    \"my_synonym_filter\"\n",
+        "                ]\n",
+        "            }\n",
+        "        },\n",
+        "        \"filter\": {\n",
+        "            \"my_synonym_filter\": {\n",
+        "                \"type\": \"synonym_graph\",\n",
+        "                \"synonyms_set\": \"my-synonyms-set\",\n",
+        "                \"updateable\": True\n",
+        "            }\n",
+        "        }\n",
+        "    }\n",
+        "}\n",
+        "\n",
+        "mappings = {\n",
+        "    \"properties\": {\n",
+        "        \"title\": {\n",
+        "            \"type\": \"text\",\n",
+        "            \"analyzer\": \"my_custom_index_analyzer\",\n",
+        "            \"search_analyzer\": \"my_custom_search_analyzer\"\n",
+        "        },\n",
+        "        \"summary\": {\n",
+        "            \"type\": \"text\",\n",
+        "            \"analyzer\": \"my_custom_index_analyzer\",\n",
+        "            \"search_analyzer\": \"my_custom_search_analyzer\"\n",
+        "        }\n",
+        "    }\n",
+        "}\n",
+        "\n",
+        "client.indices.create(index='book_index', mappings=mappings, settings=settings)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YYa3kdKvJtZW"
+      },
+      "source": [
+        "There are a few things to note in the configuration:\n",
+        "\n",
+        "- We are using the `synonym_graph` token filter.\n",
+        "- We have defined two analyzers: `my_custom_index_analyzer` and `my_custom_search_analyzer`. `my_custom_search_analyzer` is used as a [search analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-analyzer.html).\n",
+        "- `my_synonym_filter` is used only in `my_custom_search_analyzer`.\n",
+        "\n",
+        "A configuration like this is used to allow us to a) use multi-word synonyms and b) update synonyms without reindexing.\n",
+        "\n",
+        "Multi-word synonym handling is an important topic to understand if you want to leverage synonyms to their maximum benefit. See the [the subtleties of Elasticsearch synonyms](https://mauricius.dev/the-subtleties-of-elasticsearch-synonyms/) for more information."
```

Comment:
IMO I don't think we should - let's keep interlinking to Elastic related content

Reply:
Ok, I can link to [The same, but different: Boosting the power of Elasticsearch with synonyms](https://www.elastic.co/blog/boosting-the-power-of-elasticsearch-with-synonyms), but it doesn't go into nearly as much detail about token streams & multi-word synonyms. It would be nice if we had an internal doc/blog that covered that in more detail.